scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1953 @@
+        same as if the transformation was applied to the vector in the
+        following way: ``tf.apply(v) == translation + rotation.apply(v)``
+
+        Parameters
+        ----------
+        translation : array_like, shape (N, 3) or (3,)
+            A single translation vector or a stack of translation vectors.
+        rotation : `Rotation` instance
+            A single rotation or a stack of rotations.
+
+        Returns
+        -------
+        `RigidTransformation`
+            If rotation is single and translation is shape (3,), then a single
+            transformation is returned.
+            Otherwise, a stack of transformations is returned.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Creating from a single rotation and translation:
+
+        >>> t = np.array([2, 3, 4])
+        >>> r = R.from_euler("ZYX", [90, 30, 0], degrees=True)
+        >>> r.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> tf = Tf.from_components(t, r)
+        >>> tf.rotation.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> tf.translation
+        array([2., 3., 4.])
+        >>> tf.single
+        True
+
+        When applying a transformation to a vector ``v``, the result is the
+        same as if the transformation was applied to the vector in the
+        following way: ``tf.apply(v) == translation + rotation.apply(v)``
+
+        >>> r.apply([1, 0, 0])
+        array([0.       , 0.8660254, -0.5     ])
+        >>> t + r.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        >>> tf.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        """
+        return cls.from_translation(translation) * cls.from_rotation(rotation)
+
+    @cython.embedsignature(True)
+    @classmethod
+    def from_exp_coords(cls, exp_coords):
+        r"""Initialize from exponential coordinates of transformation.
+
+        This implements the exponential map that converts 6-dimensional real
+        vectors to SE(3). The first three components of the vector encode
+        the rotation and the last three components encode the translation.
+
+        The exponential coordinates of transformation can be split up into
+        a screw axis :math:`\mathcal{S} \in \mathbb{R}^6` and a scalar
+        :math:`\theta` that defines the magnitude of the displacement. For
+        pure translations, the norm of the last three components defines
+        :math:`\theta` and for all other transformations, the norm of the first
+        three components defines :math:`\theta`.
```

Comment:
This paragraph is confusing. My interpretation of exponential coordinates - conceptually the same as rotation vector in SO3, but generalized to SE3. So if we are talking about axis-magnitude, then axis is 6D vector and magnitude is computed for the whole vector.

My suggestion was to write that this exp coordinates can be technically interpreted as a literal matrix exponential for a specific 4x4 matrix (form of which I've suggested to write).

As it now stands I would just omit this paragraph.

@AlexanderFabisch help to clean up this part.

Reply:
https://github.com/scottshambaugh/scipy/pull/9